boards/arm/stm32h747i_disco/Kconfig.board

```diff
@@ -0,0 +1,16 @@
+# Kconfig - STM32H747I Discovery board configuration
+#
+# Copyright (c) 2019 Linaro Limited
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config BOARD_STM32H747I_DISCO_M7
+	bool "STM32H747I Discovery Development Board"
+	depends on SOC_STM32H747XX
+	select CPU_CORTEX_M7
```

Comment:
Weird that Board options select Cortex cores. 
could this be done in the SOC level instead?

Reply:
This was my initial attempt, but at the end, this only introduces and intermediate SOC_STM32H747XX_M7, that selects CPU_CORTEX_M7.
When electing the cortex at this level, I can remove an intermediate step which makes things clearer. Still, we're able to do cortex specific settings in soc/arm/st_stm32/stm32h7/Kconfig.soc